lib/v1.0/C/secure_pprz_transport.c

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (C) 2006  Pascal Brisset, Antoine Drouin
+ * Copyright (C) 2014-2015  Gautier Hattenberger <gautier.hattenberger@enac.fr>
+ * Copyright (C) 2017  Michal Podhradsky <mpodhradsky@galois.com>
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * @file pprzlink/secure_pprz_transport.c
+ *
+ * Building and parsing Paparazzi frames with encryption.
+ * See https://wiki.paparazziuav.org/wiki/Messages_Format
+ * for more details
+ *
+ * Pprz frame:
+ *
+ * |STX|length|... payload=(length-4) bytes ...|Checksum A|Checksum B|
+ *
+ * where checksum is computed over length and payload:
+ * @code
+ * ck_A = ck_B = length
+ * for each byte b in payload
+ *     ck_A += b;
+ *     ck_b += ck_A;
+ * @endcode
+ */
+
+#include <inttypes.h>
+#include "pprzlink/secure_pprz_transport.h"
+
+// crypto
+#include "pprzlink/hacl-c/Chacha20Poly1305.h"
```

Comment:
if HACL was a Paparazzi module, the include would be more like `#include "modules/datalink/hacl-c/..."`
Not sure what is preferable, but having the Paparazzi module makes it a little bit cleaner.